src/main/java/unibook/model/person/Person.java

```diff
@@ -68,6 +69,19 @@ public Email getEmail() {
         return Collections.unmodifiableSet(modules);
     }
 
+    /**
+     * Deletes a module from the set of modules the person has by checking through each module
+     * in the set and removing it from the set if the module code matches.
+     * @param other
+     */
+    public void removeModule(ModuleCode other) {
+        for (Module module : modules) {
+            if (module.hasModuleCode(other)) {
+                modules.remove(other);
```

Comment:
I think there might be a bug here.
Here when you call modules.remove(other) it always returns false (i.e always removes nothing from the set), because you are trying to search for a ModuleCode other in a Set of Modules. I believe you should be passing an instance of Module to remove() instead of a ModuleCode.